fuel-core/src/schema/tx.rs

```diff
@@ -42,18 +42,41 @@ impl TxQuery {
         let db = ctx.data_unchecked::<Database>();
         let key = id.0;
 
-        let tx_pool = ctx.data::<Arc<TxPool>>().unwrap();
+        let tx_status = db.get_tx_status(&key).unwrap().unwrap();
+
+        if let TransactionStatus::Submitted { .. } = tx_status {
+            let tx_pool = ctx.data::<Arc<TxPool>>().unwrap();
+
+            let found_tx = tx_pool.pool().find(&[key]).await;
 
-        let found_tx = tx_pool.pool().find(&[key]).await;
+            let return_val = Arc::try_unwrap(found_tx.get(0).unwrap().clone().unwrap()).unwrap();
```

Comment:
Why did this [regress](https://github.com/FuelLabs/fuel-core/pull/238#discussion_r836646177) back to the both ugly and hazardous code we had before I reviewed the last changes here?

While [Arc::try_unwrap](https://doc.rust-lang.org/std/sync/struct.Arc.html#method.try_unwrap) will return the inner object inside of an Arc, it has much larger consequences. The transaction pool was designed to avoid extra cloning of transactions, and also support async environments, which is why it returns all txs as an Arc. An Arc is essentially a reference counted item on the heap that is automatically dropped when no more references exist (similar to GC). Using `try_unwrap` like this could cause the API to panic (and potentially crash the entire node) if several callers requested the same tx id from the tx pool. This is because the txpool will clone the arc each time it hands out a tx, while `try_unwrap` says:

> Returns the inner value, if the Arc has exactly one strong reference.
> Otherwise, an [Err](https://doc.rust-lang.org/std/result/enum.Result.html#variant.Err) is returned with the same Arc that was passed in.

Which means that the second simultaneous transaction querier will get an error from try_unwrap, because the first caller which hasn't dropped their reference is still using it for their API response.

Reply:
Will revert as this change is touching the wrong function. I wrote it like this after I changed to loop, but I definitely agree I should've kept the old way of handling this return value.

From now on I will avoid `unwrap` unless the value can either can be guaranteed to be non-err or there is explicit error handling for when it does err